src/parsers/texttracks/webvtt/html/create_settings_attributes.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * How far from edge to start translating the text's origin,
+ * in order to keep all text inside it's box.
+ */
+import objectValues from "../../../../utils/object_values";
+
+const DEFAULT_HORIZONTAL_OFFSET = 30;
+const DEFAULT_VERTICAL_OFFSET = 5;
+
+export default function createSettingsAttributes (
+  settings : Partial<Record<string, string>>
+) : Attr {
+  const pAttr = document.createAttribute("style");
+  pAttr.value = getAttrValue(settings);
+  return pAttr;
+}
+
+const getAttrValue = (settings: Partial<Record<string, string>>) => {
+  const hasSettings = settings !== undefined && objectValues(settings).length !== 0;
+  if (!hasSettings) {
+    return "text-align:center";
+  }
+
+  return (
+    "position: absolute;" +
+    "margin: 0;" +
+    `${getTransformStyle(settings)}` +
+    `${getWidthStyle(settings.size)}` +
+    `${getPositionStyle(settings.position)}` +
+    `${getLineStyle(settings.line)}` +
+    `${getAlignStyle(settings.align)}`
+  );
+};
+
+const getTransformStyle = (settings: Partial<Record<string, string>>) => {
```

Comment:
I'm not sure I follow the need for this one.

It will only be functional when the line is centered, right ?

But even there, I'm not sure I get it. I'm not helped by my poor skill in CSS but do we not want always to translate the x coordinates -50% in those cases (as `position` will be enough for the right starting position and it will be correctly included in `width`)?

Basically we just want the declared position to be right at the middle of our text, no?

I didn't dive into `line` here yet.

Reply:
I updated this code quite a bit. Previously I was using the transforming of the cue box origin to prevent the cue box to be placed outside of the video view viewport when using positions close to 0% or 100%. This was not based upon anything in the spec, I just thought it was reasonable.

Now, when I've read the spec more carefully, I've instead considered the `position alignment` property, which basically maps to transforming the cue box's origin in the x direction. No need to "guess" a good transformation as was done before. 

So you can now be placing the cue box at the same location with different position values and position alignments, for example:

```
position:90% line-right, size:35% === position:55%, line-left, size: 35%
```

In the same manner, the `line alignment` property is now also considered, which transforms the cue box's origin in the y direction.